File: android/src/main/java/app/independo/capacitoremojipicker/core/EmojiPickerResult.java
package app.independo.capacitoremojipicker.core;

/** Result of presenting the emoji picker. */
public final class EmojiPickerResult {

    private final String emoji;

    public EmojiPickerResult(String emoji) {
        this.emoji = emoji;
    }

    /** The selected emoji, or {@code null} when the picker was dismissed without a selection. */
    public String getEmoji() {
        return emoji;
    }
}

File: ios/Sources/EmojiPicker/Core/EmojiPickerResult.swift
import Foundation

/// Result of presenting the emoji picker.
struct EmojiPickerResult {
    /// The selected emoji, or `nil` when the picker was dismissed without a selection.
    let emoji: String?
}

/// A canonical error surfaced by the service/presenter layers.
struct EmojiPickerError: Error {
    let code: String
}
